### PR TITLE
v1.3: CRUD Methods

### PR DESCRIPTION
**Pull request by srizzuto**
On this branch, I implemented the CRUD (create, update, delete) methods. All the elements of the user interface are fully functional and your application is completed.

### CODE REVIEW

File: src/addremove.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  tasks.push({ description, completed: false, index: tasks.length + 1 });
+  window.localStorage.setItem('stored', JSON.stringify(tasks));
+};
+
+export const updateIndex = (tasks) => {
+  tasks.forEach((task) => {
+    task.index = tasks.indexOf(task) + 1;
+  });
+  window.localStorage.setItem('stored', JSON.stringify(tasks));
+};
```
Comment (Dipeshtwis) [inline src/addremove.js:11]: - [ ] Please add the function to edit the task descriptions as per requirement. Currently, I am not able to edit the task.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @srizzuto, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: A function for adding a new task (add a new element to the array) and deleting a task (remove an element from the array) is implemented. Well done

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

- [ ] I would recommend fixing the `live demo link` which is pointing to the readme itself, please fix that.

**_Also check the comments under the review_**


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/addremove.js [hunk 2]
```diff
@@ -18,4 +18,10 @@ export const remove = (tasks, taskToRemove) => {
 export const removeChecked = (tasks) => {
   tasks = tasks.filter((task) => task.completed === false);
   updateIndex(tasks);
+};
+
+export const edit = (task, description) => {
+  const tasks = JSON.parse(window.localStorage.getItem('stored'));
+  tasks[task.index - 1].description = description;
+  window.localStorage.setItem('stored', JSON.stringify(tasks));
 };
```
Comment (Dipeshtwis) [inline src/addremove.js]: - [ ] When I am selecting any task to edit, it's giving me the console error that `task.index - 1` is undefined and also it doesn't update in the `localstorage` as well. Also, after refresh the updated task back into the previous state. Make sure you are using the correct index to edit.

Below is the screenshot I am attaching

Screenshot from 2021-08-28 03-05-10

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @srizzuto, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️

_Check the comments under the review._

- [ ] Also, I said the `live link` is not working, it's taking out to README itself. Please try to fix that or remove that section to make the README professional.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**

-------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by srizzuto** [comment]
- Edit error is fixed
- Live demo was removed previously. The deployment is outdated.

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @srizzuto, 

Great Work. I can edit the task description now. :clap: 
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/addremove.js (hunk 2 above)
Comment (sinansevgi) [reply src/addremove.js]: :heavy_check_mark:

**Pull request merged**